test/boost/secondary_index_test.cc

```diff
@@ -255,6 +255,9 @@ SEASTAR_TEST_CASE(test_many_columns) {
         e.execute_cql("CREATE INDEX ON tab (d)").get();
         e.execute_cql("CREATE INDEX ON tab (e)").get();
         e.execute_cql("CREATE INDEX ON tab (f)").get();
+        // #7659
+        require_rows(e, "SELECT * FROM tab WHERE d=0 AND f>0 ALLOW FILTERING", {});
```

Comment:
Don't you need eventually() here because you didn't wait for the index creation to finish?
Did this test fail before this patch?

Reply:
> Don't you need eventually() here because you didn't wait for the index creation to finish?

Apparently, _I_ don't, but that may be just a quirk of my dev environment.  I think I'll add it anyway.

> Did this test fail before this patch?

Yes, it failed in the same way as #7659.